From 5812782885624648d3d198edab5af314abadfee4 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 13 Dec 2023 13:31:00 +1000
Subject: [PATCH] oeffis: OEFFIS_DEVICE_ALL_DEVICES should translate to "all"

As the portal documentation [1] says:
  Bitmask of what device types to request remote controlling of. Default is all.

The default is only triggered if we do not submit the types at all, the
current behavior of sending a value of 0 means "none". Fix this by
skipping the "types" key if we try to select for all devices.

[1] https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.impl.portal.RemoteDesktop.html

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/liboeffis.c     |  2 +-
 test/test_oeffis.py | 24 ++++++++++++++++++++++++
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/src/liboeffis.c b/src/liboeffis.c
index 93b83176..7d2ec353 100644
--- a/src/liboeffis.c
+++ b/src/liboeffis.c
@@ -560,7 +560,7 @@ portal_select_devices(struct oeffis *oeffis)
 					&response,
 					"oa{sv}",
 					oeffis->session_path,
-					2,
+					oeffis->devices == OEFFIS_DEVICE_ALL_DEVICES ? 1 : 2,
 					"handle_token", /* string key */
 					"s", token, /* variant string */
 					"types", /* string key */
-- 
2.43.0

